### PR TITLE
Poisson reconstruction - experiments and speedup using parallel Mesh_3

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

_Describe what your pull request changes to CGAL (this can be skipped if it solves an issue already in the tracker or if it is a Feature or Small Feature submitted to the CGAL Wiki)._

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #0000, fix #0000,...
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

**Comment by sloriot** [comment]
If might be interesting to look again at #4996 (in term of runtime + still crashing)

File: Poisson_surface_reconstruction_3/examples/Poisson_surface_reconstruction_3/poisson_and_parallel_mesh_3.cpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+namespace params = CGAL::parameters;
+
+template<typename Concurrency_tag, typename PointSet>
+void poisson_reconstruction(const PointSet& points, const char* output)
+{
+  typedef CGAL::Labeled_mesh_domain_3<Kernel> Mesh_domain;
+  typedef typename CGAL::Mesh_triangulation_3<Mesh_domain, CGAL::Default, Concurrency_tag>::type Tr;
+  typedef CGAL::Mesh_complex_3_in_triangulation_3<Tr> C3t3;
+  typedef CGAL::Mesh_criteria_3<Tr> Mesh_criteria;
```
Comment (sloriot) [inline Poisson_surface_reconstruction_3/examples/Poisson_surface_reconstruction_3/poisson_and_parallel_mesh_3.cpp:52]: `warning: unused typedef 'Mesh_criteria' [-Wunused-local-typedef]`

Comment (MaelRL) [reply Poisson_surface_reconstruction_3/examples/Poisson_surface_reconstruction_3/poisson_and_parallel_mesh_3.cpp:52]: https://github.com/CGAL/cgal/pull/7891/commits/34b4b2cdf7ad7c62324a677533a4fbc5f74344a3

**Comment by sloriot** [comment]
Successfully tested in  CGAL-6.0-Ic-129

**Comment by sloriot** [comment]
Is this branch still experimental or fixes done already a good addition to have?

**Comment by MaelRL** [comment]
We have another potential improvement in the pipeline.

**Comment by sloriot** [comment]
Would that invalidate any of the fixes here?

**Comment by MaelRL** [comment]
Potentially yes, it's still related to hints.

**Comment by sloriot** [comment]
runtime errors and deprecation warnings in `Poisson_surface_reconstruction_3_Examples` of CGAL-6.0-Ic-279

**Comment by janetournois** [comment]
Warnings are due to deprecation of `Surface_mesher`. I'm taking care of it

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-282

File: Installation/CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## [Release 6.1](https://github.com/CGAL/cgal/releases/tag/v6.1)
+
+### [Poisson Surface Reconstruction](https://doc.cgal.org/6.0/Manual/packages.html#PkgPoissonSurfaceReconstruction3)
```
Comment (sloriot) [inline Installation/CHANGES.md:5]:
```suggestion
### [Poisson Surface Reconstruction](https://doc.cgal.org/6.1/Manual/packages.html#PkgPoissonSurfaceReconstruction3)
```

**Comment by sloriot** [comment]
> Potentially yes, it's still related to hints.

Any update?

**Comment by lrineau** [comment]
@MaelRL @afabri It seems this PR should have been in CGAL-6.0. Maybe it could still be a bug-fix in CGAL-6.0.1.

**Comment by sloriot** [comment]
@janetournois could you please update CHANGES.md so that it can be merged for 6.0.1?
In private discussion, @MaelRL indicated that we can merge it. There is another possible optimization with the hint but it can go into an issue.

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0.1-Ic-346

**Pull request merged**